### PR TITLE
tanh_double_grad_rules

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR types

others
### PR changes

others
### Describe

Pcard-66975
拆解tanh_double_grad

### CODE REVIEW

**Review by JiabinYang — DISMISSED** [review]
LGTM

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h [hunk 1]
```diff
@@ -125,6 +125,25 @@ void tanh_grad(const Tensor& out, const Tensor& grad_out, Tensor* grad_x) {
...
+template <typename T>
+void tanh_double_grad(const Tensor& out,
+                      const Tensor& grad_out,
+                      const Tensor& grad_x_grad,
+                      Tensor* out_grad,
+                      Tensor* grad_out_grad) {
+  // tanh grad grad : ddout = (1 - out^2) * ddx, dout = - (dout_old * 2 * out *
+  // ddx)
+  if (out_grad) {
+    auto out_grad_tmp = -1 * grad_out * 2 * out * grad_x_grad;
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: optimize this logic using tmp var

Comment (JiabinYang) [reply paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: one comment

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h [hunk 2]
```diff
@@ -125,6 +125,25 @@ void tanh_grad(const Tensor& out, const Tensor& grad_out, Tensor* grad_x) {
...
+    set_output<T>(out_grad_tmp, out_grad);
+  }
+
+  if (grad_out_grad) {
+    auto grad_out_grad_tmp = (1 - out * out) * grad_x_grad;
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: like 137

**Review by JiabinYang — COMMENTED** [review]
some commments

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Pull request merged**